From e99b987e459f83d7a96a0f3586cfab8833cc0810 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Tue, 19 Jan 2016 16:03:04 +0800
Subject: [PATCH] atomisp2: kernel configuration change align with BTNS kernel

CONFIG_EXTERNAL_CAMERA_ICGOTM and CONFIG_PLATFORM_BTNS are
going to be deprecated and they are being replaced by
CONFIG_EXTERNAL_BTNS_CAMERA.

Original-Tracked_On: #H1504148602
Change-Id: I788aef1419b4c08ab82a970f2533bb2f47613d99
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/96411
Reviewed-by: DaleiX Liu <daleix.liu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: Dessert M BXT RVP (SC)
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Build: Dessert M CHT FFD (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-16657
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/461753
---
 drivers/external_drivers/camera/Kconfig                           | 2 +-
 drivers/external_drivers/camera/Makefile                          | 2 +-
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.c        | 8 ++++----
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c       | 2 +-
 .../drivers/media/pci/atomisp2/css2400b0_v21_build/Makefile       | 2 +-
 5 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/drivers/external_drivers/camera/Kconfig b/drivers/external_drivers/camera/Kconfig
index c2e8418..19ff3e9 100644
--- a/drivers/external_drivers/camera/Kconfig
+++ b/drivers/external_drivers/camera/Kconfig
@@ -2,7 +2,7 @@ if EXTERNAL_CAMERA
 source drivers/external_drivers/camera/drivers/Kconfig
 endif
 
-if EXTERNAL_CAMERA_ICGOTM
+if EXTERNAL_BTNS_CAMERA
 # for BTNS camera drivers
 source drivers/external_drivers/camera/drivers/Kconfig
 endif
diff --git a/drivers/external_drivers/camera/Makefile b/drivers/external_drivers/camera/Makefile
index 9411a2c..83723b5 100644
--- a/drivers/external_drivers/camera/Makefile
+++ b/drivers/external_drivers/camera/Makefile
@@ -12,4 +12,4 @@
 #
 
 obj-$(CONFIG_EXTERNAL_CAMERA) += drivers/
-obj-$(CONFIG_EXTERNAL_CAMERA_ICGOTM) += drivers/
+obj-$(CONFIG_EXTERNAL_BTNS_CAMERA) += drivers/
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
index c34c727..8d8bb0e 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.c
@@ -29,7 +29,7 @@
 #else
 #include <media/v4l2-chip-ident.h>
 #endif
-#ifdef CONFIG_PLATFORM_BTNS
+#ifdef CONFIG_EXTERNAL_BTNS_CAMERA
 #include "ov8858_btns.h"
 #else
 #include "ov8858.h"
@@ -1675,7 +1675,7 @@ static int __update_ov8858_device_settings(struct ov8858_device *dev,
 					   u16 sensor_id)
 {
 	if (sensor_id == OV8858_CHIP_ID)
-#ifdef CONFIG_PLATFORM_BTNS
+#ifdef CONFIG_EXTERNAL_BTNS_CAMERA
 		dev->vcm_driver = &ov8858_vcms[OV8858_ID_DEFAULT];
 #else
 		dev->vcm_driver = &ov8858_vcms[OV8858_SUNNY];
@@ -1933,7 +1933,7 @@ static int ov8858_g_ctrl(struct v4l2_ctrl *ctrl)
 	case V4L2_CID_EXPOSURE_ABSOLUTE:
 		ctrl->val = dev->exposure;
 		break;
-#ifdef CONFIG_PLATFORM_BTNS
+#ifdef CONFIG_EXTERNAL_BTNS_CAMERA
 	case V4L2_CID_LINK_FREQ:
 		ctrl->val = 360000000;
 		break;
@@ -2259,7 +2259,7 @@ static const struct v4l2_ctrl_config ctrls[] = {
 		.max = V4L2_EXPOSURE_APERTURE_PRIORITY,
 		.step = 1,
 	},
-#ifdef CONFIG_PLATFORM_BTNS
+#ifdef CONFIG_EXTERNAL_BTNS_CAMERA
 	{
 		.ops = &ctrl_ops,
 		.id = V4L2_CID_LINK_FREQ,
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 4ecde58..a4b545f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -3239,7 +3239,7 @@ void atomisp_apply_css_parameters(
 	   before AIQ for ISP2.7 is available,	we have to keep such
 	   workaround for a reasonable image color
 	 */
-#ifdef CONFIG_PLATFORM_BTNS
+#ifdef CONFIG_EXTERNAL_BTNS_CAMERA
 	{
 		static struct atomisp_css_ob2_config imx227_ob2_config =
 			{1024, 1024, 1024, 1024};
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21_build/Makefile b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21_build/Makefile
index 3c5496d..f36ce00 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21_build/Makefile
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21_build/Makefile
@@ -15,7 +15,7 @@ ccflags-y += -DATOMISP_POSTFIX=\"css2400b0_v21\" -DISP2400B0
 ISP2400B0 := y
 postfix := css2400b0_v21
 
-ifeq ($(CONFIG_PLATFORM_BTNS),y)
+ifeq ($(CONFIG_EXTERNAL_BTNS_CAMERA),y)
 ccflags-y += -DATOMISP_FWNAME=\"shisp_merr.bin\"
 endif
 
-- 
1.9.1

